core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -175,14 +175,19 @@ class KafkaApis(val requestChannel: RequestChannel,
   }
 
   def handleUpdateMetadataRequest(request: RequestChannel.Request) {
-    val updateMetadataRequest = request.requestObj.asInstanceOf[UpdateMetadataRequest]
-
-    authorizeClusterAction(request)
+    val correlationId = request.header.correlationId
+    val updateMetadataRequest = request.body.asInstanceOf[UpdateMetadataRequest]
 
-    replicaManager.maybeUpdateMetadataCache(updateMetadataRequest, metadataCache)
+    val responseHeader = new ResponseHeader(correlationId)
```

Comment:
Nitpick: it would be better to move this to after `updateMetadataResponse` as we only need it when sending the response.